1.18.2 bump dependencies

SrRapero720 opened the pull request.

PaintNinja reviewed (changes requested):
We don't intentionally make breaking changes after an RB and you seemed to have bumped deps without checking for this. Please make sure you only update deps to versions that don't have any breaking changes

SrRapero720 commented:
> We don't intentionally make breaking changes after an RB and you seemed to have bumped deps without checking for this. Please make sure you only update deps to versions that don't have any breaking changes

i was testing the bump eventbus but is really broken on basic modpacks, i reverted it, other libs seems to work well

PaintNinja commented:
I'm afraid you'll need to do more than just "I tested basic modpacks". Avoid any major version bumps in general and look through the commit history and changelogs of the things you're bumping to make sure you don't accidentally break stuff.

LexManos commented:
Is there any particular reason you wish for these dependencies to be bumped? Is there something you're wanting that the old versions don't have? AT's bump removes all the ANTLR stuff, which is not functionally breaking but transitively breaking.
Coremods and ASM should be fine as they are backwards compatible..
Installer is fine as its only install time, so doesn't effect mods.

SrRapero720 commented:
> I'm afraid you'll need to do more than just "I tested basic modpacks". Avoid any major version bumps in general and look through the commit history and changelogs of the things you're bumping to make sure you don't accidentally break stuff.

(rewording) i tested EventBus bump but it is a hard breaking change even on basic modpacks, i reverted it.
But, other dependencies seems to works well on small and big mdpacks

Basic modpack is composed by basic server-side, client-side mods such as FancyMenu, ModernFix, Embeddium, Oculus, Dimensional Threading, Create and misc mods

Big modpack is at least 238 mods. with some unstable mods like Canary or Pluto
![image](https://github.com/MinecraftForge/MinecraftForge/assets/29090346/e25fbf6d-9ea6-4e41-b268-3979b6b7aa63)

Result
![image](https://github.com/MinecraftForge/MinecraftForge/assets/29090346/14cff64b-0a10-47dd-9578-dcbe86849533)
![image](https://github.com/MinecraftForge/MinecraftForge/assets/29090346/18c9287c-2436-4e47-9cc0-a9326efb0623)

modernfix forces load all mixins, so... there is no error

SrRapero720 commented:
Coremods, ASM and ATs are just for bug fixing (take advantage of this PR without having to make another unnecessary PR)
Installer bump has the purpose of make more easier release updates without rewrite the installer (and of course, make possible test PRs in prod)

i am doing the same for 1.19.2 and 1.20.1, but i started with 1.18.2 because is the most old time consuming version.

i will keep the eventbus thing for another chance, because needs more deep testing, some transitional changes of 5.x to 6.x aren't hard breaking, and the most relevant change (for me) is this: https://github.com/MinecraftForge/EventBus/commit/eb8e549bbf75dd3c478924615e34390e383efc45

SrRapero720 commented:
all versions PRs are done, no transitively breaking changes

SrRapero720 committed:
bump dependencies

- AT 8.1.1
- Coremods 5.1.3
- Eventbus 5.0.7
- ASM 9.6
- Installer 2.2.+
- Installer Tools 1.4.1

PaintNinja commented on build.gradle:
```diff
@@ -55,14 +55,14 @@ ext {
-    ACCESSTRANSFORMERS_VERSION = '8.0.4'
-    COREMODS_VERSION = '5.0.1'
-    EVENTBUS_VERSION = '5.0.3'
+    ACCESSTRANSFORMERS_VERSION = '8.1.4'
```
```suggestion
    ACCESSTRANSFORMERS_VERSION = '8.1.1'
```
8.1.2 removes ANTLR. To use that version or higher, you need to make sure the same ANTLR version it used to use is still bundled with Forge to ensure no intentional breaking changes. You're better off just bumping it to 8.1.1 and leaving future bumps to it as another PR.

PaintNinja commented on buildSrc/src/main/groovy/net/minecraftforge/forge/tasks/checks/CheckATs.groovy:
```diff
@@ -171,7 +171,7 @@ abstract class CheckATs extends CheckTask {
 
     private static List<String> joinBack(TreeMap<String, ATParser.Entry> lines, Map<String, InheritanceData> inheritance, IMappingFile mappings) {
-        final data = []
+        final data = [] as List<String>
```
```suggestion
        final List<String> data = []
```

SrRapero720 replied:
thats how was done in 1.20.x branh, but fair enough

PaintNinja commented on buildSrc/src/main/groovy/net/minecraftforge/forge/tasks/checks/CheckATs.groovy:
```diff
@@ -307,4 +307,4 @@ class ATParser {
         }
     }
-}
+}
```
Likely accidental line removal

SrRapero720 committed:
‎‎‎

SrRapero720 committed:
revert AT bump

SrRapero720 committed:
bump coremods to 5.1.2

PaintNinja commented on build.gradle:
```diff
@@ -56,13 +56,13 @@ ext {
-    COREMODS_VERSION = '5.0.1'
-    EVENTBUS_VERSION = '5.0.3'
+    COREMODS_VERSION = '5.1.2'
+    EVENTBUS_VERSION = '5.0.7'
```
```suggestion
    EVENTBUS_VERSION = '5.0.3'
```
https://maven.minecraftforge.net/net/minecraftforge/eventbus/5.0.7/eventbus-5.0.7-changelog.txt
No changes for consumers going from 5.0.3 to 5.0.7, so please leave it as-is so that the cached version from old installs can still be used to keep updates fast on slow connections.

SrRapero720 committed:
revert eventbus bump

Co-authored-by: Paint_Ninja <PaintNinja@users.noreply.github.com>
